sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java

```diff
@@ -237,27 +200,27 @@ public void setDataValues(List<Object> dataValues) {
     this.dataValues = dataValues;
   }
 
-  public BeamSqlRowType getDataType() {
+  public BeamRecordType getDataType() {
     return dataType;
   }
 
-  public void setDataType(BeamSqlRowType dataType) {
+  public void setDataType(BeamRecordType dataType) {
     this.dataType = dataType;
   }
 
-  public void setNullFields(List<Integer> nullFields) {
-    this.nullFields = nullFields;
+  public BitSet getNullFields() {
+    return nullFields;
   }
 
-  public List<Integer> getNullFields() {
-    return nullFields;
+  public void setNullFields(BitSet nullFields) {
```

Comment:
I think so, the information is MUST to decode as we skip null values when encoding.
--Calculate it on the fly is possible although, a trade-off between a loop-scan and store a BitSet.